src/BenchmarksServer/Startup.cs

```diff
@@ -328,6 +398,38 @@ void CleanJob()
             }
         }
 
+        private static (string containerId, string imageName) DockerBuildAndRun(string path, ServerJob job, string hostname)
+        {
+            var source = job.Source;
+            // Docker image names must be lowercase
+            var imageName = $"{Path.GetDirectoryName(source.Project)}_{Guid.NewGuid()}".ToLowerInvariant();
+            var cloneDir = Path.Combine(path, Git.Clone(path, source.Repository));
+
+            if (!string.IsNullOrEmpty(source.BranchOrCommit))
+            {
+                Git.Checkout(cloneDir, source.BranchOrCommit);
+            }
+
+            ProcessUtil.Run("docker", $"build -t {imageName} -f {source.DockerFile} .", cloneDir);
+
+            var result = ProcessUtil.Run("docker", $"run -d -p {job.Port}:{job.Port} {imageName}");
```

Comment:
This should use `--network host` instead of `-p port:port` for improved performance.  Also, if you add `--rm`, then the container will be automatically removed after it's stopped, so you will no longer need `docker container rm`.